cfme/tests/infrastructure/test_publish_vm_to_template.py

```diff
@@ -19,8 +19,12 @@ def vm_crud(provider):
     collection = provider.appliance.provider_based_collection(provider)
     vm_name = random_vm_name(context='pblsh')
     vm = collection.instantiate(vm_name, provider)
-    deploy_template(vm.provider.key, vm_name, provider.data['small_template'],
-                    timeout=2500)
+    if provider.one_of(VMwareProvider):
+        deploy_template(vm.provider.key, vm_name,
+            provider.data['templates']['small_template']['name'], timeout=2500)
+    else:
+        deploy_template(vm.provider.key, vm_name, provider.data['small_template'],
+                        timeout=2500)
```

Comment:
**Required** If a provider does not have these key, it raise a `KeyError`. We can handle this two ways:
1) Use a try/except
2)  Break out getting the template name and use a `.get()`, setting it to none if the key is not found. Then check to make sure the template name is not none.